UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarning/Admin/RadioNavigationalWarningAdminTest.cs

```diff
@@ -0,0 +1,167 @@
+            Assert.AreEqual("RnwAdminListReferance", adminListFilter.RadioNavigationalWarningsAdminList[2].Reference);
+            Assert.AreEqual(new DateTime(2022, 1, 1), adminListFilter.RadioNavigationalWarningsAdminList[2].DateTimeGroup);
+            Assert.AreEqual("RnwAdminListSummary", adminListFilter.RadioNavigationalWarningsAdminList[2].Summary);
+            Assert.AreEqual("RnwAdminListContent", adminListFilter.RadioNavigationalWarningsAdminList[2].Content);
+            Assert.AreEqual(new DateTime(2099, 1, 1), adminListFilter.RadioNavigationalWarningsAdminList[2].ExpiryDate);
+            Assert.AreEqual("No", adminListFilter.RadioNavigationalWarningsAdminList[2].IsDeleted);
+            Assert.IsNotNull(((ViewResult)result).ViewData["WarningTypes"]);
+            Assert.IsNotNull(((ViewResult)result).ViewData["Years"]);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithWarningTypeFilter_ThenReturnFilteredListAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 20;
+            IActionResult result = await _controller.Index(1, 1, null);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(4, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual(1, adminListFilter.PageCount);
+            Assert.AreEqual(0, adminListFilter.SrNo);
+            Assert.AreEqual(1, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithYearFilter_ThenReturnFilteredListAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 20;
+            IActionResult result = await _controller.Index(1, null, 2020);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(2, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual(1, adminListFilter.PageCount);
+            Assert.AreEqual(0, adminListFilter.SrNo);
+            Assert.AreEqual(1, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithWarningTypeAndYearFilter_ThenReturnFilteredListAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 20;
+            IActionResult result = await _controller.Index(1, 2, 2024);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(1, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual(1, adminListFilter.PageCount);
+            Assert.AreEqual(0, adminListFilter.SrNo);
+            Assert.AreEqual(1, adminListFilter.CurrentPageIndex);
```

Comment:
Same as above, can we also check the warning type filter applied correctly for expected id and name. Also if possible check expected year data is returned.

Reply:
Changes are done as requestee, Thank You.